src/main/kotlin/io/emeraldpay/dshackle/upstream/CurrentMultistreamHolder.kt

```diff
@@ -16,120 +16,41 @@
  */
 package io.emeraldpay.dshackle.upstream
 
-import io.emeraldpay.dshackle.cache.CachesEnabled
-import io.emeraldpay.dshackle.cache.CachesFactory
-import io.emeraldpay.dshackle.startup.UpstreamChange
-import io.emeraldpay.dshackle.upstream.bitcoin.BitcoinMultistream
-import io.emeraldpay.dshackle.upstream.bitcoin.BitcoinUpstream
-import io.emeraldpay.dshackle.upstream.calls.CallMethods
-import io.emeraldpay.dshackle.upstream.calls.DefaultBitcoinMethods
-import io.emeraldpay.dshackle.upstream.calls.DefaultEthereumMethods
-import io.emeraldpay.dshackle.upstream.ethereum.EthereumMultistream
-import io.emeraldpay.dshackle.upstream.ethereum.EthereumPosMultiStream
-import io.emeraldpay.dshackle.upstream.ethereum.EthereumPosUpstream
-import io.emeraldpay.dshackle.upstream.ethereum.EthereumUpstream
-import io.emeraldpay.grpc.BlockchainType
 import io.emeraldpay.grpc.Chain
 import org.slf4j.LoggerFactory
-import org.springframework.beans.factory.annotation.Autowired
-import org.springframework.stereotype.Repository
+import org.springframework.stereotype.Component
 import reactor.core.publisher.Flux
 import reactor.core.publisher.Sinks
-import java.util.Collections
-import java.util.concurrent.Callable
+import java.util.*
 import java.util.concurrent.ConcurrentHashMap
 import java.util.concurrent.locks.ReentrantLock
 import javax.annotation.PreDestroy
 import kotlin.concurrent.withLock
 
-@Repository
+@Component
 open class CurrentMultistreamHolder(
-    @Autowired private val cachesFactory: CachesFactory
+    private val multistreams: List<Multistream>
 ) : MultistreamHolder {
 
     private val log = LoggerFactory.getLogger(CurrentMultistreamHolder::class.java)
 
-    private val chainMapping = ConcurrentHashMap<Chain, Multistream>()
+    private val chainMapping = ConcurrentHashMap<Chain, Multistream>().apply {
+        multistreams.forEach { this[it.chain] = it }
+    }
     private val chainsBus = Sinks.many()
         .multicast()
```

Comment:
sinks not working any more since you are not emit anything - I would suggest to remove and switch to event. Probably UpstreamUpdateEvent